### PR TITLE
feat: Add max_value_length option

### PR DESCRIPTION
**Pull request by HazAT**

### CODE REVIEW

File: tests/Serializer/SerializerTest.php [hunk 1]
```diff
@@ -95,11 +96,45 @@ public function testNull(bool $serializeAllObjects): void
...
+            $result = $this->invokeSerialization($serializer, $input);
+
+            $this->assertInternalType('string', $result);
+            $this->assertLessThanOrEqual(500, \strlen($result));
+        }
+    }
+
+    public function testClippingUTF8Characters(): void
+    {
+        if (!\extension_loaded('mbstring')) {
```
Comment (ste93cry) [inline tests/Serializer/SerializerTest.php]: Since we're requiring the `mbstring` extension in the `composer.json` we can safely remove this code

File: tests/Serializer/SerializerTest.php [hunk 2]
```diff
@@ -95,11 +96,45 @@ public function testNull(bool $serializeAllObjects): void
...
+        $this->assertSame(JSON_ERROR_NONE, json_last_error());
+    }
+
     /**
+     * @param Options $options
+     *
      * @return Serializer
      */
-    protected function createSerializer(): AbstractSerializer
+    protected function createSerializer(Options $options = null): AbstractSerializer
```
Comment (ste93cry) [inline tests/Serializer/SerializerTest.php]: Since this is a `nullable` argument with a default value it should be `?Options $options = null`

File: src/Options.php [hunk 3]
```diff
@@ -597,6 +597,28 @@ public function setDefaultIntegrations(bool $enable): void
...
+     *
+     * @return int
+     */
+    public function getTruncationLength(): int
+    {
+        return $this->options['truncation_length'];
+    }
+
+    /**
+     * Sets the truncation length for specific values in the event payload.
```
Comment (ste93cry) [inline src/Options.php]: Does this really apply to only some values? I thought it was applied to everything that is processed by the serializers

File: tests/Serializer/AbstractSerializerTest.php [hunk 4]
```diff
@@ -317,20 +317,6 @@ public function testLongString(bool $serializeAllObjects): void
         }
     }
 
-    public function testLongStringWithOverwrittenMessageLength(): void
```
Comment (Jean85) [inline tests/Serializer/AbstractSerializerTest.php]: Why did you move those?

File: src/Serializer/AbstractSerializer.php [hunk 5]
```diff
@@ -76,15 +74,22 @@ abstract class AbstractSerializer
...
 
-    public function __construct(int $maxDepth = 3, ?string $mbDetectOrder = null, int $messageLimit = Client::MESSAGE_MAX_LENGTH_LIMIT)
+    /**
+     * AbstractSerializer constructor.
+     *
+     * @param Options     $options       The SDK configuration options
+     * @param int         $maxDepth
+     * @param string|null $mbDetectOrder
+     */
+    public function __construct(Options $options, int $maxDepth = 3, ?string $mbDetectOrder = null)
```
[the remark this answers is not in the document]
Comment (ste93cry) [reply src/Serializer/AbstractSerializer.php]: Does it really makes sense to have it as a client option? It seems to be something related to the serializer, imho we should drop it and just rely on the argument of this constructor instead

File: src/Options.php (hunk 3 above)
Comment (HazAT) [reply src/Options.php]: I quickly went over the code, we do not use this value for everything.
Maybe we should tackle this in a different PR.

File: tests/Serializer/AbstractSerializerTest.php (hunk 4 above)
Comment (HazAT) [reply tests/Serializer/AbstractSerializerTest.php]: Because the tests didn't run.

File: tests/EventFactoryTest.php [hunk 6]
```diff
@@ -16,20 +16,29 @@
...
 class EventFactoryTest extends TestCase
 {
+    /**
+     * @var Options
+     */
+    private $options;
+
+    protected function setUp(): void
+    {
+        $this->options = new Options();
```
Comment (ste93cry) [inline tests/EventFactoryTest.php]: I don't see a real value in having this instance set as variable of the class, is there any reason you moved it here?

Comment (HazAT) [reply tests/EventFactoryTest.php]: Mostly for convenience, having `new Options()` in almost every test doesn't really make sense. Also, we've done this in other test classes.

File: tests/Serializer/AbstractSerializerTest.php (hunk 4 above)
Comment (ste93cry) [reply tests/Serializer/AbstractSerializerTest.php]: Wasn't it possible to just add the `$options` parameter to the function?

File: tests/EventFactoryTest.php (hunk 6 above)
Comment (ste93cry) [reply tests/EventFactoryTest.php]: Not that it changes much, but only two tests are using the options so I don't think we should move the instantiation to the `setUp` method which creates an additional object each time a test case is executed even if there is no need for it
